azkaban-common/src/main/java/azkaban/database/AbstractJdbcLoader.java

```diff
@@ -16,19 +16,27 @@
 
 package azkaban.database;
 
+import azkaban.event.Event;
+import azkaban.event.EventHandler;
+import azkaban.event.EventListener;
+import azkaban.event.MultitonListenerSet;
+import azkaban.metrics.CommonMetrics;
+import azkaban.utils.Props;
+
 import java.io.IOException;
 import java.sql.Connection;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.ArrayList;
+import java.util.HashSet;
 
 import org.apache.commons.dbutils.DbUtils;
 import org.apache.commons.dbutils.QueryRunner;
 import org.apache.commons.dbutils.ResultSetHandler;
 
-import azkaban.utils.Props;
 
-public abstract class AbstractJdbcLoader {
+public abstract class AbstractJdbcLoader implements EventHandler {
```

Comment:
Do you think the JDBC Loader is playing too many roles here?
What are your thoughts on this? https://en.wikipedia.org/wiki/Composition_over_inheritance